### PR TITLE
Add end-to-end rpm build targets

### PR DESCRIPTION
**Pull request by fierlion**
### Summary
This adds make targets to build a dockerfree pause container, dockerfree cni-plugins, dockerfree agent image and generic-rpm-integrated and amazon-linux-rpm-integrated targets which build all of the above into an rpm.

### Implementation details
Targets which are architecture dependent will introspect the build host's architecture (`uname -m`).  We'll need to continue to rely on the host arch rather than cross-compiling, because we don't have a clean way to cross compile c code for the pause container.  
```
dockerfree-pause
dockerfree-certs
dockerfree-cni-plugins
dockerfree-agent-image
```
and the rpm targets:
```
amazon-linux-rpm-integrated
generic-rpm-integrated
```
In order to build with go 1.15 - go 1.17, I reverted a change to remove the old style build syntax.  
Note all new targets are additive and do not collide with existing targets.

### Testing
I tested all of the above targets manually on both amd64 and arm64 hosts with go 1.15, 1.16 and 1.17

New tests cover the changes: no

### Description for the changelog
Add end to end rpm build targets.

### Licensing

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -356,6 +357,36 @@ get-deps-init:
...
+	cp packaging/amazon-linux-ami-integrated/ecs.conf ecs.conf
+	cp packaging/amazon-linux-ami-integrated/ecs.service ecs.service
+	cp packaging/amazon-linux-ami-integrated/amazon-ecs-volume-plugin.conf amazon-ecs-volume-plugin.conf
+	cp packaging/amazon-linux-ami-integrated/amazon-ecs-volume-plugin.service amazon-ecs-volume-plugin.service
+	cp packaging/amazon-linux-ami-integrated/amazon-ecs-volume-plugin.socket amazon-ecs-volume-plugin.socket
+	tar -czf ./sources.tgz ecs-init scripts misc agent amazon-ecs-cni-plugins amazon-vpc-cni-plugins agent-container VERSION
+	test -e SOURCES || ln -s . SOURCES
+	rpmbuild --define "%_topdir $(PWD)" -bb ecs-init.spec
+	find RPMS/ -type f -exec cp {} . \;
+	touch .rpm-done
```
Comment (sparrc) [inline Makefile:372]: i think this should be `touch .amazon-linux-rpm-integrated-done` ?

File: Makefile [hunk 2]
```diff
@@ -356,6 +357,36 @@ get-deps-init:
...
+	./scripts/update-version.sh
+	cp packaging/generic-rpm-integrated/amazon-ecs-init.spec amazon-ecs-init.spec
+	cp packaging/generic-rpm-integrated/ecs.service ecs.service
+	cp packaging/generic-rpm-integrated/amazon-ecs-volume-plugin.service amazon-ecs-volume-plugin.service
+	cp packaging/generic-rpm-integrated/amazon-ecs-volume-plugin.socket amazon-ecs-volume-plugin.socket
+	tar -czf ./sources.tgz ecs-init scripts misc agent amazon-ecs-cni-plugins amazon-vpc-cni-plugins agent-container VERSION
+	test -e SOURCES || ln -s . SOURCES
+	rpmbuild --define "%_topdir $(PWD)" -bb amazon-ecs-init.spec
+	find RPMS/ -type f -exec cp {} . \;
+	touch .rpm-done
```
Comment (sparrc) [inline Makefile:386]: and `touch .generic-rpm-integrated-done` ?

File: Makefile [hunk 3]
```diff
@@ -370,7 +401,22 @@ get-deps-init:
...
+	./scripts/update-version.sh
+	cp packaging/generic-rpm/amazon-ecs-init.spec amazon-ecs-init.spec
+	cp packaging/generic-rpm/ecs.service ecs.service
+	cp packaging/generic-rpm/amazon-ecs-volume-plugin.service amazon-ecs-volume-plugin.service
+	cp packaging/generic-rpm/amazon-ecs-volume-plugin.socket amazon-ecs-volume-plugin.socket
+	tar -czf ./sources.tgz ecs-init scripts
+	test -e SOURCES || ln -s . SOURCES
+	rpmbuild --define "%_topdir $(PWD)" -bb amazon-ecs-init.spec
+	find RPMS/ -type f -exec cp {} . \;
+	touch .rpm-done
```
Comment (sparrc) [inline Makefile:414]: this touch doesn't really do anything anymore unless the make target is split into `.rpm-done` and `rpm-done: .rpm-done` like the integrated ones are

File: scripts/build-cni-plugins [hunk 4]
```diff
@@ -18,15 +18,37 @@
 
 set -ex
 
+architecture=""
+case $(uname -m) in
+    x86_64)
+	architecture="amd64"
+        ;;
+    aarch64)
```
Comment (prateekchaudhry) [inline scripts/build-cni-plugins:26]: I get "arm64" for "uname -m" on my arm laptop, so maybe we can consider adding that as well

File: scripts/build-agent-image [hunk 5]
```diff
@@ -22,6 +22,22 @@ ROOT=$( cd "$( dirname "${BASH_SOURCE[0]}" )/.." && pwd )
 cd "${ROOT}"
 AGENT_VERSION=$(cat VERSION)
 
+architecture=""
+case $(uname -m) in
+    x86_64)
+        architecture="amd64"
+        ;;
+    aarch64)
```
Comment (prateekchaudhry) [inline scripts/build-agent-image:30]: here as well, can consider adding "arm64"

File: Makefile [hunk 6]
```diff
@@ -370,7 +401,22 @@ get-deps-init:
...
+
+generic-rpm: .generic-rpm-done
+
+
+dockerfree-all: dockerfree-agent-image generii-rpm
```
Comment (prateekchaudhry) [inline Makefile:419]: is "generii-rpm" a typo?

Comment (fierlion) [reply Makefile:419]: yeah -- vim miss.  fixing

**Review by sparrc — APPROVED** [review]

**Review by angelcar — APPROVED** [review]

**Review by YashdalfTheGray — APPROVED** [review]

**Pull request merged**